### ISSUE TITLE
Why is Tinyproxy reporting HTTP 500 right after getting DNS response?

### ISSUE THREAD

**Issue by Proofer** [tinyproxy/tinyproxy#445]
## Tinyproxy version
1.11.0

## Issue
Probably my issue, so I'm just asking for help. Trying to set up a web proxy server on my LAN that provides a WireGuard VPN tunnel to a VPN server in the cloud. I'm using Firefox as the client and have configured it to use the proxy server that is running Tinyproxy. On an attempt to access a site, Firefox immediately reports "having trouble finding ... can't connect." Using the tshark sniffer to see traffic, I see the following, where "..." indicates I've truncated a line and "!" indicates a line that tshark displayed out of timestamp order and which I have here moved into timestamp order.  
Addresses:  
192.168.50.200 client (Firefox)
192.168.50.212 Tinyproxy server and WireGuard client machine
10.4.0.2 WireGuard tunnel virtual interface `wg0`
10.4.0.1, referenced in .conf `Bind`, is the cloud end of tunnel; 10.4.0.0/16 is routed to wg0 interface
1.1.1.2  DNS server in cloud

````
  502 47.114586667 192.168.50.200 → 192.168.50.212 HTTP 277 CONNECT ifconfig.me:443 HTTP/1.1
  503 47.114636300 192.168.50.212 → 192.168.50.200 TCP 66 8888 → 57666 [ACK] ...
  504 47.115387336     10.4.0.2 → 1.1.1.2      DNS 57 Standard query 0x7a6e A ifconfig.me
  505 47.115428139     10.4.0.2 → 1.1.1.2      DNS 57 Standard query 0x8f64 AAAA ifconfig.me
 !515 47.319108832      1.1.1.2 → 10.4.0.2     DNS 128 Standard query response 0x8f64 AAAA ifconfig.me ...
 !516 47.319109241      1.1.1.2 → 10.4.0.2     DNS 73 Standard query response 0x7a6e A ifconfig.me A 34.117.59.81
  506 47.319604790 192.168.50.212 → 192.168.50.200 TCP 170 HTTP/1.0 500 Unable to connect  [TCP segment of a reassembled PDU]
````
What have I done, or omitted, to cause Tinyproxy to report an HTTP 500 error immediately after receiving a DNS response?

.conf file:
Port 8888
Listen 192.168.50.212
Bind 10.4.0.1
Timeout 600
DefaultErrorFile "/usr/share/tinyproxy/default.html"
StatFile "/usr/share/tinyproxy/stats.html"
Syslog On
LogLevel Info
MaxClients 2
Allow 127.0.0.1
Allow 192.168.0.0/16
ViaProxyName "tinyproxy"

**Comment by rofl0r** [comment]
my guess is that from bind interface 34.117.59.81 can't be reached

in doubt, run tinyproxy under strace -f (in foreground)

**Comment by Proofer** [comment]
All educated guesses welcome! ... Fom the TinyProxy/Wireguard machine's terminal:
```
vpnpeer:~$ wget ifconfig.me
--2022-06-18 17:21:48--  http://ifconfig.me/
Resolving ifconfig.me (ifconfig.me)... 34.117.59.81
Connecting to ifconfig.me (ifconfig.me)|34.117.59.81|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 12 [text/plain]
Saving to: ‘index.html’

index.html                                   100%[====...>]      12  --.-KB/s    in 0s      

2022-06-18 17:21:49 (575 KB/s) - ‘index.html’ saved [12/12]

vpnpeer:~$ cat index.html
12.34.56.79vpnpeer:~$ 
```
ifconfig.me is a "what is my ip address" type of web service. In the last line I see the public IP address of the remote WireGuard server (which I have here replaced with "12.34.56.78" for security reasons). So doesn't that imply that 34.117.59.81 can be reached from the bind interface, which is the local end of the WireGuard tunnel? (Unfamiliar with strace, but I will pounce on it right after dinner.)

**Comment by rofl0r** [comment]
wget uses the default outgoing interface. i'd bet if you remove the bind directive for tinyproxy, it will work too.

**Comment by Proofer** [comment]
Yeah! Removed .conf's Bind and I saw ifconfig.me's output in Firefox on my Mac! Just as in my prior issue, I was overthinking the .conf directives. Thanks again!

Then I got ambitious and, still in proxy mode, tried to come here to post the previous sentence; but Firefox timed out on this github page. I'll be  poring over the tshark output from that...
